jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -116,12 +134,212 @@ MlirToJlmConverter::ConvertOperation(
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::sge)
+    {
+      return rvsdg::node_output::node(rvsdg::bitsge_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::sgt)
+    {
+      return rvsdg::node_output::node(rvsdg::bitsgt_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::sle)
+    {
+      return rvsdg::node_output::node(rvsdg::bitsle_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::slt)
+    {
+      return rvsdg::node_output::node(rvsdg::bitslt_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::uge)
+    {
+      return rvsdg::node_output::node(rvsdg::bituge_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::ugt)
+    {
+      return rvsdg::node_output::node(rvsdg::bitugt_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::ule)
+    {
+      return rvsdg::node_output::node(rvsdg::bitule_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else if (ComOp.getPredicate() == ::mlir::arith::CmpIPredicate::ult)
+    {
+      return rvsdg::node_output::node(rvsdg::bitult_op::create(
+          (size_t)ComOp.getOperandTypes()[0].cast<::mlir::IntegerType>().getWidth(),
+          inputs[0],
+          inputs[1]));
+    }
+    else
+    {
+      auto message = util::strfmt(
+          "Comparision Predicate not implemented:",
+          stringifyCmpIPredicate(ComOp.getPredicate()).str(),
+          "\n");
+      JLM_UNREACHABLE(message.c_str());
+    }
+  }
+
+  /* #region Arithmetic Integer Operation*/
+  // TODO try to replace the big else if copying of code with a template or something cleaner
+  //! Here the LLVM dialect where only implemented for AddOp. Other operation should maybe be
+  //! imported Need to choose which one of mlir::arith or mlir::LLVM to use for the MLIR
+  //! representation
+  else if (auto andOp = ::mlir::dyn_cast<::mlir::LLVM::AddOp>(&mlirOperation))
```

Comment:
Maybe move the entire handling of the binary operations into its own function and factor out the common code?